### PR TITLE
Dev java - Declarations

### PR DESCRIPTION
**Pull request by marijakatic**
Declarations Java AST Builder and tests.

### CODE REVIEW

**Review by ivan-ristovic — CHANGES_REQUESTED** [review]
This is excellent! The code is extremely readable and precise! 

I have added a few comments, most of them regarding the stylepoints, I would suggest running automatic formatting from within Visual Studio, it will use formatting rules from `.editorconfig` so you will probably fix most of my comments just by running that.

Thank you for the excellent work! :)

File: LINVAST.Imperative/Builders/Java/JavaASTBuilder.Declarations.cs [hunk 1]
```diff
@@ -15,5 +14,499 @@ namespace LINVAST.Imperative.Builders.Java
     public sealed partial class JavaASTBuilder : JavaBaseVisitor<ASTNode>, IASTBuilder<JavaParser>
     {
 
+        // package and import declarations:
```
[the remark this answers is not in the document]
Comment (marijakatic) [reply LINVAST.Imperative/Builders/Java/JavaASTBuilder.Declarations.cs]: Wow, great, thank you!

File: LINVAST.Imperative/Nodes/Common/Modifiers.cs [hunk 2]
```diff
@@ -97,5 +105,7 @@ public enum QualifierFlags
         Static = 1,
         Const = 2,
         Volatile = 4,
+        Default = 8,
+        Final = 16,
```
[the remark this answers is not in the document]
Comment (marijakatic) [reply LINVAST.Imperative/Nodes/Common/Modifiers.cs]: Ok. I fixed it. :)

File: LINVAST.Tests/Imperative/Builders/Java/InterfaceDeclarationTests.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+{
+    internal sealed class InterfaceDeclarationTests : DeclarationTestsBase
+    {
+
+        [Test]
+        public void EmptyBodyInterfaceDeclTest()
+        {
+            string src1 = "interface Interface1 {}";
+            TypeDeclNode ast1 = this.GenerateAST(src1).As<TypeDeclNode>();
+            StringAssert.IsMatch(@"Interface1\s*{\s*}\s*", ast1.GetText());
```
[the remark this answers is not in the document]
Comment (marijakatic) [reply LINVAST.Tests/Imperative/Builders/Java/InterfaceDeclarationTests.cs]: Great remark. I should have thought about that.
Thank you! :)

**Review by ivan-ristovic — APPROVED** [review]
LGTM, waiting for others to do their work and we can merge.

**Pull request merged**